controllers/glance_controller.go

```diff
@@ -886,7 +893,17 @@ func (r *GlanceReconciler) generateServiceConfig(
 ) error {
 	labels := labels.GetLabels(instance, labels.GetGroupLabel(glance.ServiceName), serviceLabels)
 
-	ospSecret, _, err := secret.GetSecret(ctx, h, instance.Spec.Secret, instance.Namespace)
+	// ensure a MariaDBAccount CR and Secret exist.
+	// this mariadb API function will as an interim step actually generate a
+	// new MariaDBAccount and Secret if one does not exist already.   in a
+	// future release, this function may change to emit an error if the
+	// MariaDBAccount was not already created ahead of time (e.g. by openstack-operator
+	// or end-user YAML declaration)
+	databaseAccount, dbSecret, err := mariadbv1.EnsureMariaDBAccount(
```

Comment:
note this no longer is given a mariadb username; as these usernames are expected to rotate regularly, EnsureMariaDBAccount also generates the username now, as it already generates the name of the Secret and the password.

Reply:
i added this so that you can exercise a full create new mariadbaccount + update pods to use new username/password + remove finalizers from old one just by going in the CR and changing "DatabaseAccount" to any random new string